src/SphinxEngine.php

```diff
@@ -225,8 +225,16 @@ protected function performSearch(Builder $builder)
 
         $query = $this->sphinx
             ->select('*', SphinxQL::expr('WEIGHT() AS weight'))
-            ->from($index)
-            ->match('*', SphinxQL::expr('"' . str_replace('"', '\"', $builder->query) . '"/1.0'));
+            ->from($index);
+
+        if ($builder->callback) {
```

Comment:
Just a query:
Will there be a case where `->match()` and `call_user_func()` both will be necessary?

Reply:
if both is necessary, we could do that both in the  callback.